Objects/moduleobject.c

```diff
@@ -866,6 +866,40 @@ _Py_module_getattro(PyModuleObject *m, PyObject *name)
     return _Py_module_getattro_impl(m, name, 0);
 }
 
+static int
+module_setattro(PyModuleObject *m, PyObject *name, PyObject *value)
+{
+    PyObject *setattr, *delattr;
+    assert(m->md_dict != NULL);
+    PyErr_Clear();
+    if (value == NULL) {
+        delattr = PyDict_GetItemWithError(m->md_dict, &_Py_ID(__delattr__));
+        if (PyErr_Occurred()) {
+            return -1;
+        }
+        if (delattr) {
+            PyObject_CallFunctionObjArgs(delattr, name, NULL);
```

Comment:
wait, you cannot discard the result: you must Py_DECREF() it, or you will leak memory.

Same remark for the second PyObject_CallFunctionObjArgs() call below.